sub-packages/bionemo-esm2/src/bionemo/esm2/model/model.py

```diff
@@ -162,6 +162,9 @@ def __init__(
                 seq_len_interpolation_factor=seq_len_interpolation_factor,
             )
 
+        self.config.num_layers = (
+            self.config.num_layers // self.config.pipeline_model_parallel_size
+        ) * self.config.pipeline_model_parallel_size
```

Comment:
leaving a breadcrumb to the slack thread here: https://nvidia.slack.com/archives/C074Z808N05/p1744835028471649?thread_ts=1744834381.782379&cid=C074Z808N05

let's make sure we don't resize the model and instead find a way to handle PP with uneven numbers of layers